mapit/add_areas_from_file/core.py

```diff
@@ -119,6 +122,34 @@ class Error(Exception):
 
 
 def run(filename, parameters, logger=None):
+    if zipfile.is_zipfile(filename):
+        if logger:
+            logger.info("Unzipping %s" % filename)
+
+        zipped = zipfile.ZipFile(filename)
+        root_shp_files = [
+            fn
+            for fn in zipped.namelist()
+            if fn.endswith(".shp") and fn.count(os.sep) == 1
```

Comment:
I think this should be a `0` (or I've created a weird zip file) because I think this will match files in the first subdirectory.

On macOS I created a zip by right-clicking the `parish_region.{dbf,prj,shp,shx}` files in Finder and selecting "Compress":

<img width="399" alt="image" src="https://user-images.githubusercontent.com/4776/212765494-0513f62d-25c4-423a-9e7e-c18ed3efe3d6.png">

Which resulted in the following archive as described by `zipinfo`:

```
$ zipinfo parish_region.zip
Archive:  parish_region.zip
Zip file size: 114295453 bytes, number of entries: 8
-rw-r--r--  2.0 unx  5680219 bX defN 22-May-18 07:55 parish_region.dbf
-rw-r--r--  2.0 unx      255 bX defN 22-May-18 07:55 __MACOSX/._parish_region.dbf
-rw-r--r--  2.0 unx      417 bX defN 22-May-18 07:53 parish_region.prj
-rw-r--r--  2.0 unx      212 bX defN 22-May-18 07:53 __MACOSX/._parish_region.prj
-rw-r--r--  2.0 unx 240030836 bX defN 22-May-18 07:55 parish_region.shp
-rw-r--r--  2.0 unx      311 bX defN 22-May-18 07:55 __MACOSX/._parish_region.shp
-rw-r--r--  2.0 unx   115132 bX defN 22-May-18 07:55 parish_region.shx
-rw-r--r--  2.0 unx      255 bX defN 22-May-18 07:55 __MACOSX/._parish_region.shx
8 files, 245827637 bytes uncompressed, 114293823 bytes compressed:  53.5%
```

You can see the `__MACOSX` junk that it creates for whatever reason is being matched as the first `.shp` file by the current code, which crashes:

```
INFO:mapit_import:Unzipping /Users/davea/Desktop/parish_region.zip
INFO:mapit_import:Found __MACOSX/._parish_region.shp
INFO:mapit_import:Importing from /var/folders/fg/v3p1rmqj7ms7n4nlvn14cn5w0000gp/T/tmpgb5mwpt3/__MACOSX/._parish_region.shp
INFO:mapit_import:(will not save to db as 'commit' not specified)
Traceback (most recent call last):
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/contrib/gis/gdal/datasource.py", line 71, in __init__
    ds = capi.open_ds(force_bytes(ds_input), self._write, byref(ds_driver))
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/contrib/gis/gdal/prototypes/errcheck.py", line 127, in check_pointer
    raise GDALException('Invalid pointer returned from "%s"' % func.__name__)
django.contrib.gis.gdal.error.GDALException: Invalid pointer returned from "OGROpen"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/davea/Code/mapit/mapit/manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/core/management/base.py", line 571, in handle
    label_output = self.handle_label(label, **options)
  File "/Users/davea/Code/mapit/mapit/mapit/management/commands/mapit_import.py", line 242, in handle_label
    add_areas_from_file.run(filename, parameters, logger)
  File "/Users/davea/Code/mapit/mapit/mapit/add_areas_from_file/core.py", line 153, in run
    ds = DataSource(filename)
  File "/Users/davea/.virtualenvs/mapit/lib/python3.10/site-packages/django/contrib/gis/gdal/datasource.py", line 75, in __init__
    raise GDALException('Could not open the datasource at "%s"' % ds_input)
django.contrib.gis.gdal.error.GDALException: Could not open the datasource at "/var/folders/fg/v3p1rmqj7ms7n4nlvn14cn5w0000gp/T/tmpgb5mwpt3/__MACOSX/._parish_region.shp"
```

Changing to a `0` works as expected. If your zip files work with `1` I'd be interested in how you created them and what `zipinfo` tells you? Might need to change how you find files in the root of the archive to be more flexible.

Reply:
Good spot - I was zipping a folder containing the `.shp` rather than zipping the files directly. Have updated to support either scenario.